xsplice: Don't require a .xsplice.depends section

We don't know how xsplice payloads are going to be deployed
so don't make unnecessary requirements.

diff --git a/xen/common/xsplice.c b/xen/common/xsplice.c
index 11e0e30..16c6609 100644
--- a/xen/common/xsplice.c
+++ b/xen/common/xsplice.c
@@ -678,7 +678,6 @@ static int check_special_sections(struct payload *payload,
 {
     unsigned int i;
     static const char *const names[] = { ".xsplice.funcs" ,
-                                         ".xsplice.depends",
                                          ".note.gnu.build-id"};
 
     for ( i = 0; i < ARRAY_SIZE(names); i++ )
@@ -800,7 +799,9 @@ static int find_special_sections(struct payload *payload,
         spin_unlock(&payload_lock);
     }
 
+    payload->dep.len = 0;
     sec = xsplice_elf_sec_by_name(elf, ".xsplice.depends");
+    if ( sec && sec->sec->sh_size )
     {
         n = (Elf_Note *)sec->load_addr;
         if ( sec->sec->sh_size <= sizeof *n )
@@ -1008,7 +1009,7 @@ static int load_payload_data(struct payload *payload, uint8_t *raw, ssize_t len)
 static int apply_depcheck(struct payload *payload)
 {
     if ( !payload->dep.len )
-        return -EINVAL;
+        return 0;
 
     if ( list_empty(&applied_list) )
     {
